Subject: Efika has no legacy IO
From: olh@suse.de

Efika has no legacy IO, fixes kbd warnings like:

i8042.c: No controller found.

---
 arch/powerpc/platforms/52xx/efika.c |    7 +++++++
 1 file changed, 7 insertions(+)

Index: b/arch/powerpc/platforms/52xx/efika.c
===================================================================
--- a/arch/powerpc/platforms/52xx/efika.c
+++ b/arch/powerpc/platforms/52xx/efika.c
@@ -186,6 +186,12 @@ static void efika_show_cpuinfo(struct se
 	of_node_put(root);
 }
 
+static int efika_check_legacy_ioport(unsigned int baseport)
+{
+	printk(KERN_WARNING "XXX rejected access to port %u\n", baseport);
+	return -ENODEV;
+}
+
 static void __init efika_setup_arch(void)
 {
 	rtas_initialize();
@@ -272,5 +278,6 @@ define_machine(efika)
 	.get_boot_time		= rtas_get_boot_time,
 	.calibrate_decr		= generic_calibrate_decr,
 	.phys_mem_access_prot	= pci_phys_mem_access_prot,
+	.check_legacy_ioport	= efika_check_legacy_ioport,
 };
 
